sklearn/decomposition/kernel_pca.py

```diff
@@ -124,6 +124,7 @@ class KernelPCA(BaseEstimator, TransformerMixin):
         component analysis. In Advances in kernel methods,
         MIT Press, Cambridge, MA, USA 327-352.
     """
+    _centerer = KernelCenterer()
```

Comment:
This needs to be fit per instance. Storing it on the class means that doesn't work. Rather it needs to be assigned in `fit` instead of `__init__`.